pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -349,6 +352,100 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         emit SwapEnabledSet(swapEnabled);
     }
 
+    /**
+     * @dev Remove a token from the pool. This is a permissioned function that performs the following operations:
+     * - Verify there are enough tokens, and there is no ongoing weight change
+     * - Calculate the BPT value of the full token balance, and returns it for possible use by the caller
+     * - Exit the pool, returning the full balance of the token to the recipient
+     * - Deregister the token in the Vault (since it now has zero balance)
+     * - Reduce the total weight by the weight of the token removed, and update the number of tokens and `_tokenState`
+     *   (all other weights then scale accordingly)
+     */
+    function removeToken(uint256 tokenIndex, address recipient) external authenticate whenNotPaused returns (uint256) {
+        return _removeToken(tokenIndex, recipient);
+    }
+
+    function _removeToken(uint256 tokenIndex, address recipient) internal returns (uint256) {
+        // Calculate the BPT value of the full token balance (other values returned to avoid duplication)
+        //(uint256 bptAmountIn, uint256 minAmountOut, uint256 normalizedWeight) = getTotalBptForToken(tokenIndex);
+        // Exit the pool, returning the full balance of the token to the recipient
+        (IERC20[] memory tokens, uint256[] memory rawBalances, ) = getVault().getPoolTokens(getPoolId());
+        _require(tokenIndex < tokens.length, Errors.OUT_OF_BOUNDS);
+        _require(tokens.length > 2, Errors.MIN_TOKENS);
+        // Verify there is no ongoing weight change
+        _require(
+            0 == _calculateWeightChangeProgress() || FixedPoint.ONE == _calculateWeightChangeProgress(),
+            Errors.REMOVE_TOKEN_DURING_WEIGHT_CHANGE
+        );
+
+        IERC20 removedToken = tokens[tokenIndex];
+        uint256 weightOfRemovedToken = _exitWithEntireBalance(
+            tokens,
+            _getNormalizedWeights(),
+            tokenIndex,
+            _scalingFactor(removedToken),
+            rawBalances[tokenIndex],
+            recipient
+        );
+
+        // Deregister the token in the Vault
+        IERC20[] memory tokensToRemove = new IERC20[](1);
+        tokensToRemove[0] = removedToken;
+        getVault().deregisterTokens(getPoolId(), tokensToRemove);
+
+        emit TokenRemoved(removedToken);
+
+        // Clean up data structures and update the token count
+        delete _tokenState[removedToken];
+        _setMiscData(_getMiscData().insertUint7(tokens.length - 1, _TOTAL_TOKENS_OFFSET));
+
+        // Decrease the total weight by the weight of the token being removed
+        _totalWeight -= weightOfRemovedToken;
+
+        // The bpt amount corresponding to the token is simply its proportional share of the totalSupply
+        return weightOfRemovedToken.mulDown(totalSupply());
```

Comment:
We need to multiply by the normalized weight, not the raw one. And we need to normalize using the old weight sum (prior to removal).

Reply:
It is doing that - it's calling _getNormalizedWeights before doing the remove or adjusting the total